service/routes.py

```diff
@@ -366,6 +366,56 @@ def create_product_in_wishlist(wishlist_id):
     status.HTTP_201_CREATED
   )
 
+######################################################################
+# RENAME A WISHLIST
+######################################################################
+@app.route("/wishlists/<int:wishlist_id>", methods=["PUT"])
+def rename_a_wishlist(wishlist_id):
+  """
+  Renames a wishlist
+  This endpoint will rename an existing wishlist
+  """
+  app.logger.info("Request to rename the wishlist with id %s"\
+    % (wishlist_id))
+
+  request_data = request.get_json()
+
+  wishlist = Wishlist.find_by_id(wishlist_id)
+
+  if not request_data['name']:
+    return make_response(
+      jsonify(
+        data=[],
+        message = "Field name cannot be null"
+      ),
+      status.HTTP_400_BAD_REQUEST
+    )
+
+  query_res = Wishlist.find_all()
+
+  for i in query_res:
+    if request_data['name']==i.name:
```

Comment:
I'm not sure we should throw an error in this case. Two users can easily both have a "Christmas wishlist". We identify a wishlist by id, so there should not be any errors if there are two wishlists with the same name. Maybe you should compare the names only for the same user? Even in that case, I would let it be, but if you'd like, you can implement the following logic: if a user has a wishlist with the name A, and this request tries to rename a different  wishlist to the same name, you make it A 1, A 2 etc, kind of like your OS does when it sees several files/folders with the same name.

But this is probably overkill 🤷‍♀️

Reply:
I think this makes sense, and I will revise it.

Reply:
I used this logic: when there exists a wishlist have the same name and user_id, change the new wishlist's "name" to "name 2". This is the same way that macos handles.